public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,578 @@
+                .call(d3.axisBottom(x)
+                    .tickSize(-Height)
+                    .tickFormat("")
+                )
+
+            // Rotate axis ticks
+            xAxis.selectAll("text")
+                .style("text-anchor", "end")
+                .attr("dx", "-.8em")
+                .attr("dy", ".15em")
+                .attr("transform", "rotate(-65)");
+
+            // Add X axis label
+            svg.append("text")
+                .attr("text-anchor", "end")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Date (dd:hh:m)");
+
+            // Add Y axis label
+            svg.append("text")
+                .attr("transform", "rotate(-90)")
+                .attr("y", 0 - Margin.left)
+                .attr("x", 0 - Height / 2)
+                .attr("dy", "1em")
+                .style("text-anchor", "middle")
+                .text("Memory Utilization(GB)");
+
+            let decimalFormatter = d3.format(".2s");
+            let yLimit = data[0].memory_usage.total;
+            // Add Y axis
+            let y = d3.scaleLinear()
+                .domain([0, yLimit])
+                .range([ Height, 0 ]);
+            svg.append("g").call(d3.axisLeft(y).ticks(5).tickFormat((d) => decimalFormatter(d).replace('G', 'GB')))
+
+            // Add the Y gridlines
+            svg.append("g")			
+                .attr("class", "memoryGrid")
+                .call(d3.axisLeft(y)
+                    .tickSize(-Width)
+                    .tickFormat("")
+                )
+
+            // Add a clipPath: everything out of this area won't be drawn.
+            let clip = svg.append("defs").append("svg:clipPath")
+                .attr("id", "clip")
+                .append("svg:rect")
+                .attr("width", Width )
+                .attr("height", Height )
+                .attr("x", 1)
+                .attr("y", 0);
+
+            // Add brushing
+            let brush = d3.brushX()                 // Add the brush feature using the d3.brush function
+                .extent( [ [0,0], [Width, Height] ] ) // initialise the brush area: start at 0,0 and finishes at width,height: it means I select the whole graph area
+                .on("end", updateChart) // Each time the brush selection changes, trigger the 'updateChart' function
+
+            // Create the scatter variable: where both the circles and the brush take place
+            let areaChart = svg.append('g')
+                .attr("clip-path", "url(#clip)")
+
+            // Create an area generator
+            let area = d3.area().x(d => x(d.datetime)).y0(y(0)).y1(d => y(d.memory_usage.used))
+
+            // Add the brushing
+            areaChart
+                .append("g")
+                .attr("class", "brush")
+                .call(brush);
```

Comment:
Otherwise this feature is really great! Have you considered adding it to the traffic graphs too? It would be really helpful to be able to zoom in on those, and I think this would work as a much nicer alternative to the existing time-frame drop-down.

Reply:
Yes, I am planning implementing it.